presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -114,6 +114,50 @@ public void testCreateFunction()
         assertQueryFails("CREATE FUNCTION testing.test.foo(x varchar) RETURNS varchar LANGUAGE UNSUPPORTED EXTERNAL", "Catalog testing does not support functions implemented in language UNSUPPORTED");
     }
 
+    @Test
+    public void testCreateFunctionWithCoercion()
+    {
+        assertQuerySucceeds("CREATE FUNCTION testing.test.return_double() RETURNS DOUBLE RETURN 1");
+        String createFunctionReturnDoubleFormatted = "CREATE FUNCTION testing.test.return_double ()\n" +
+                "RETURNS DOUBLE\n" +
+                "COMMENT ''\n" +
+                "LANGUAGE SQL\n" +
+                "NOT DETERMINISTIC\n" +
+                "CALLED ON NULL INPUT\n" +
+                "RETURN CAST(1 AS double)";
+
+        MaterializedResult rows = computeActual("SHOW CREATE FUNCTION testing.test.return_double()");
+        assertEquals(rows.getMaterializedRows().get(0).getFields(), ImmutableList.of(createFunctionReturnDoubleFormatted, ""));
+
+        rows = computeActual("SELECT testing.test.return_double() + 1");
+        assertEquals(rows.getMaterializedRows().get(0).getFields().get(0), 2.0);
+
+        assertQuerySucceeds("CREATE FUNCTION testing.test.return_varchar() RETURNS VARCHAR RETURN 'ABC'");
+        String createFunctionReturnVarcharFormatted = "CREATE FUNCTION testing.test.return_varchar ()\n" +
+                "RETURNS varchar\n" +
+                "COMMENT ''\n" +
+                "LANGUAGE SQL\n" +
+                "NOT DETERMINISTIC\n" +
+                "CALLED ON NULL INPUT\n" +
+                "RETURN CAST('ABC' AS varchar)";
+
+        rows = computeActual("SHOW CREATE FUNCTION testing.test.return_varchar()");
+        assertEquals(rows.getMaterializedRows().get(0).getFields(), ImmutableList.of(createFunctionReturnVarcharFormatted, ""));
```

Comment:
Similar to what we're doing with the previous test case, let's also verify the execution of the function. e.g
e.g.
```
computeActual("SELECT lower(testing.test.return_varchar())", ...
```

Reply:
add the unit test case